style: ruff format

zzstoatzz opened the pull request.

desertaxle commented on src/integrations/prefect-redis/prefect_redis/lease_storage.py:
```diff
@@ -40,6 +44,69 @@ def _expiration_key(self, lease_id: UUID) -> str:
+            end
+            return 1
+            """
+            self._create_sha = await self.redis_client.script_load(create_script)
```
Can we use `register_script`? TIL, that makes it so we won't need to send the script over the wire each time, which boosts performance.

desertaxle commented on src/integrations/prefect-redis/prefect_redis/lease_storage.py:
```diff
@@ -166,12 +251,37 @@ async def renew_lease(self, lease_id: UUID, ttl: timedelta) -> None:
     async def revoke_lease(self, lease_id: UUID) -> None:
         try:
             lease_key = self._lease_key(lease_id)
+            # Read lease to clean up indexes
+            data = await self.redis_client.get(lease_key)
+            holder_key: str | None = None
+            resource_ids: list[UUID] = []
+            if data is not None:
+                lease = self._deserialize_lease(data)
+                resource_ids = lease.resource_ids
+                holder = (
+                    getattr(lease.metadata, "holder", None) if lease.metadata else None
+                )
+                if hasattr(holder, "model_dump"):
+                    holder = holder.model_dump(mode="json")  # type: ignore[attr-defined]
+                holder_key = self._holder_key(holder)
```
Can you include the read operation in the revoke Lua script? There's a possible race condition here if the lease key is removed between read and the revoke Lua script.

desertaxle commented on src/integrations/prefect-redis/tests/test_lease_storage.py:
```diff
@@ -307,3 +307,71 @@ async def test_holder_round_trip(self, storage: ConcurrencyLeaseStorage):
+        setattr(meta, "holder", holder)
+
+        lease = await storage.create_lease([rid1, rid2], ttl, meta)
+        assert {
+            "type": holder["type"],
+            "id": holder["id"],
+        } in await storage.list_holders_for_limit(rid1)
```
Pretty sure these assertions can be simplified?
```suggestion
        assert holder in await storage.list_holders_for_limit(rid1)
```

zzstoatzz commented:
Thanks, Alex! Addressed:

- Switched to  for both create/revoke Lua scripts. The call sites now use small async wrappers so we never ship the script body and avoid the  branching.
- Moved the lease read into the revoke Lua script. The script now s the lease JSON, decodes with , derives  and , and conditionally removes the holder indexes only if the mapping still points to the same lease id. This eliminates the race between a client-side GET and the revoke call.
- Simplified the tests to assert  /  per your suggestion.

Let me know if you want me to similarly fold resource-id handling into the create script for symmetry; current create still takes precomputed keys but avoids extra RTT via .
